### PR TITLE
Python 3 compatibility fixes

### PR DESCRIPTION
**Pull request by pietrodn**
This PR contains all the changes needed to make the tests pass under Python 3 (together with a [patch](https://github.com/ArduPilot/pymavlink/pull/141) in pymavlink).

Python 2 compatibility should be maintained (the tests continue to pass in Python 2).

Per @peterbarker advice, this PR contains *only* the Python 3-specific fixes from PR #784, which is superseded. The other fixes are in #788.

### CODE REVIEW

File: dronekit/__init__.py [hunk 1]
```diff
@@ -1316,8 +1316,12 @@ def listener(self, name, msg):
                         self._params_last = monotonic.monotonic()
                         self._params_duration = start_duration
                     self._params_set[msg.param_index] = msg
-                self._params_map[msg.param_id] = msg.param_value
-                self._parameters.notify_attribute_listeners(msg.param_id, msg.param_value,
+
+                # Normalize parameter name (remove trailing null bytes and convert to string)
+                param_name = msg.param_id.rstrip(b'\x00').decode('utf8')
```
Comment (peterbarker) [inline dronekit/__init__.py]: I think we really, really need to address this on the pymavlink side :-)

**Comment by pietrodn** [comment]
Done!

**Comment by pietrodn** [comment]
Don't merge it yet — I have some changes to make.

**Comment by pietrodn** [comment]
Ok, [I moved](https://github.com/ArduPilot/pymavlink/pull/141) the encoding of the string parameters of MAVLink messages into the code generation part of pymavlink.
That should be cleaner than before.

**Comment by mobhuyan** [comment]
I am eagerly waiting for this. Can someone give me an idea when will it make it as release?

**Comment by tpwrules** [comment]
@pietrodn 
Hey, I'm not entirely sure where to put this, but there is at least one more change that needs to be made to make the program actually work.
On line 1293 of `dronekit/__init__.py`, `self._master.mav.param_request_read_send(0, 0, '', i)` needs to be `self._master.mav.param_request_read_send(0, 0, bytes(), i)`.
Otherwise, the program randomly hangs with no explanation when the listener takes control.

**Comment by pietrodn** [comment]
@mobhuyan you can clone my branch on the forked repo. The main problem is that you will also need a py3-compatible pymavlink. So, you will also need https://github.com/ArduPilot/pymavlink/pull/141

**Comment by pietrodn** [comment]
@peterbarker I think the PR is ready for merge: the changes here are minimal — since most Python 3 bugs are to be fixed in pymavlink — and everything should be Python2-compatible.
Is there anything else to fix in this changeset?

**Comment by peterbarker** [comment]
Merged, thanks!

**Comment by hamishwillee** [comment]
@peterbarker @mrpollo I just got a build warning for docs off this. A reminder that we need to get auto-building of docs fixed. I don't have visibility of the build server, so I can't look at this.

**Pull request closed without merging**